### ISSUE TITLE
Startup Command Field

### ISSUE THREAD

**Issue by pauljohnstone** [Azure/wordpress-linux-appservice#139]
Has there been any further issue with the Startup Command Field?
I need to apply rewrite rules, which from my understanding need to be applied to nginx.conf 

**Test rewrite rule below:**
`server {
    listen 80;
    server_name your-domain.azurewebsites.net; 
    location /Lib/abc/ {
        rewrite ^/Lib/abc/(.*)$ /Lib/ABC/$1 break;
    }
}`

The startup.sh file is located in the /home/dev/  when reviewing the file it's no script at all, which makes me think that the script is not being applied from the startup command field and is being ignored.

If this is still the case then what is the workaround solution?

**Note on the following link**
https://github.com/Azure/wordpress-linux-appservice/blob/main/WordPress/running_post_startup_scripts.md

Note: As of September 2022, the startup script MUST exist at (/home/dev/startup.sh). Further, the value of the Startup Command field in the Azure Portal found under [App Service for Linux] > Configuration > General Settings > Startup Command is ignored. The Azure team is aware of this and working to make this flow more natural.

**Comment by samrueby** [comment]
> The startup.sh file is located in the /home/dev/ when reviewing the file it's no script at all, which makes me think that the script is not being applied from the startup command field and is being ignored.

I am trying to understand "it's no script at all". By default, it's an empty file. It's empty because it's free for you to use. If you put commands in this file, they will execute when the container boots.

**Comment by pauljohnstone** [comment]
Hi @samrueby,

Apologies for any confusion. I've been trying to clarify the process for applying a script within the startup command field in the Azure portal. It seems that when you apply a script, it should ideally write to the current `startup.sh` file located under the `home/dev` folder.

As you mentioned, manual addition of the script to the `startup.sh` file is necessary for it to take effect. Additionally, copying the `nginx default.conf` file to the `/home` directory is required for applying updates. Consequently, the `startup.sh` script will handle copying and replacing any changes to the nginx directory. 

Here's a snippet of the script:

`echo "Custom script"
cp /home/default.conf /etc/nginx/conf.d/default.conf
/usr/sbin/nginx -s reload`

_Note that the echo statement has be added to review logs to confirm whether or not the startup script is being executed._

In recent weeks, upon review, I discovered that the current issue with this particular image lies in how supervisord is configured, which explains why the startup script when applied to the startup command field is being ignored.

**Comment by samrueby** [comment]
Ah, yes. On that same page is documented: 
>Further, the value of the Startup Command field in the Azure Portal found under [App Service for Linux] > Configuration > General Settings > **Startup Command is ignored**. The Azure team is aware of this and working to make this flow more natural.

So it's a known issue.

Glad like you were able to get through the issue.

**Comment by KloudstackHosting** [comment]
Question: Has the Startup Command bug been fixed?

If no script is applied to the Startup Command field, will it still look up /home/dev/startup.sh?
I do not think this is the case.

And I do not see that the Startup Command script is being saved to this local eg /home/dev/

**Comment by samrueby** [comment]
The value in the Azure Portal under Configuration > General Settings > Startup Command is still ignored.

The container will still execute /home/dev/startup.sh on startup. I.e., if a script file exists on the file system at /home/dev/startup.sh, it will be executed.

**Comment by KloudstackHosting** [comment]
Yeah, that's what I thought, but I was running some tests…the startup.sh file was not being executed on app restart.
Added the script within the Startup Command field, and it applied.

**Comment by samrueby** [comment]
I recall another issue where you posted where the output gave a hint that you may not be running the official WordPress container on App Service. See the image below.

![Image](https://github.com/user-attachments/assets/b8051321-b03e-4f16-8e46-c03d053ce79f)

Does your App Service configuration show you are using the WordPress stack?

**Comment by KloudstackHosting** [comment]
Was that on https://github.com/Azure/wordpress-linux-appservice/issues/198

Did have some permission issues and it came down to a webjob running in the background, when clearing the nginx cache.
Still reviewing the issue with applying cache rules.


WordPress PHP 8.3
mcr.microsoft.com/appsvc/wordpress-debian-php:8.3

`#!/bin/bash
# Copy updated configuration files
cp /home/custom-spec-settings.conf /etc/nginx/conf.d/spec-settings.conf
cp /home/default.conf /etc/nginx/conf.d/default.conf
# Reload NGINX
/usr/sbin/nginx -s reload`

**Comment by samrueby** [comment]
Hi- did you resolve the issue?

**Comment by KloudstackHosting** [comment]
Hi @samrueby 

Yes, work it out…

The W3TC plugin generates rules that are not applied to the default Nginx setup.

The best way to apply them is to add an `include /home/site/wwwroot/nginx.conf;` to the default.conf file just before the XMLRPC block.

Set up your startup.sh script to apply the default.conf file on restart.

I’m currently working on a script—a webjob—that will apply changes automatically when alterations are made within the W3TC plugin.

This will then include the rules that W3TC generates for the browser cache.
The other option would be to control it from Front Door rules, but that's not ideal for the customer; it’s better to handle it within the WordPress admin.

Regarding the Startup Command Field under configuration settings, I’ve removed all scripts for now.

**Comment by KloudstackHosting** [comment]
Hi @samrueby 

Regarding the Startup Command Field under configuration settings, I’ve removed all scripts for now.
Running all scripts within the startup.sh file under the /dev folder at current.
